src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/LinuxUtilizationProvider.cs

```diff
@@ -91,7 +92,7 @@ public double CpuPercentage()
 
                     if (deltaHost > 0 && deltaCgroup > 0)
                     {
-                        var percentage = Math.Min(Hundred, deltaCgroup / deltaHost * _scale);
+                        var percentage = Math.Min(1.0, deltaCgroup / deltaHost * _scale);
```

Comment:
Curious, why do we need the `Min` logic here? (sounds like a sequencing issue where different variables are fetched at different time, or is this about clock skew)

Reply:
IIRC sometimes you can exceed your service's CPU limits in a container. @rafal-mz is it the case?

Reply:
That is correct that those values are read sequentially (one after another), so how we read them might affect the precision. We might also lose a bit of precision while multiplying by fraction of cpu (cgroups allow to have 1.5 cpu for instance).  Also, in calculations we are assuming that all cpus produce the same number of cycles through the given time slice, which is probably not the case.

I didn't look for the root cause of the issue, since I've seen that other tools (like docker stats) have the same problem. As, it is impossible to spend more resources than we have and this is some approxximation error, I've cut it not to trick the customers.

Reply:
Thanks for the explanations!